packages/sdk-communication-layer/src/services/SocketService/ChannelManager/setupChannelListeners.ts

```diff
@@ -94,9 +64,62 @@ export function setupChannelListeners(
   const { socket } = instance.state;
   const { keyExchange } = instance.state;
 
-  socketEventListenerMap.forEach(({ event, handler }) => {
-    socket?.io.on(event, () => handler(instance));
-  });
+  if (instance.state.setupChannelListeners) {
+    console.warn(
+      `SocketService::${instance.state.context}::setupChannelListener socket listeners already set up for channel ${channelId}`,
+    );
+  }
+
+  // Only available for the originator -- used for connection recovery
+  if (socket && instance.state.isOriginator) {
```

Comment:
Is there a reason we went back to a clean approach to this?

 socketEventListenerMap.forEach(({ event, handler }) => {
    socket?.io.on(event, () => handler(instance));
  });

Reply:
actually I realized that the io socket recovery are optional and duplicate the recovery process already handled by the on disconnect event. We can actually remove all the io.on events and just keep them for debug for now